### PR TITLE
Platformer "Almost Final" version.

### PR DESCRIPTION
**Pull request by JVeg199X**
Still need to add enemies to 4-1 and 4-2, but those are exclusively data changes.

### CODE REVIEW

File: main/modes/platformer/entity.c [hunk 1]
```diff
@@ -568,9 +476,20 @@ void despawnWhenOffscreen(entity_t *self)
...
-        (self->x >> SUBPIXEL_RESOLUTION) > (self->tilemap->mapOffsetX + TILEMAP_DISPLAY_WIDTH_PIXELS + DESPAWN_THRESHOLD) ||
-        (self->y >> SUBPIXEL_RESOLUTION) < (self->tilemap->mapOffsetY - DESPAWN_THRESHOLD) ||
-        (self->y >> SUBPIXEL_RESOLUTION) > (self->tilemap->mapOffsetY + TILEMAP_DISPLAY_HEIGHT_PIXELS + DESPAWN_THRESHOLD))
+        (self->x >> SUBPIXEL_RESOLUTION) > (self->tilemap->mapOffsetX + TILEMAP_DISPLAY_WIDTH_PIXELS + DESPAWN_THRESHOLD)
+    )
+    {
+        destroyEntity(self, true);
+    }
+
+    if (self->y > 63616){
```
Comment (AEFeinstein) [inline main/modes/platformer/entity.c:485]: This is an oddly specific number

File: main/modes/platformer/platformer_sounds.h [hunk 2]
```diff
@@ -0,0 +1,2378 @@
...
+//==============================================================================
+#include <stdint.h>
+#include "musical_buzzer.h"
+
+//==============================================================================
+// Constants
+//==============================================================================
+
+//by FuryfulFawful
+static const song_t bgmDemagio =
```
Comment (AEFeinstein) [inline main/modes/platformer/platformer_sounds.h:15]: I really, really, really don't like variables in headers, even `const`. Put these in a C file, put the `extern` declarations in the header. And if you haven't merge the latest `main` into `platformer` and double check that the jukebox still works right.

File: main/modes/platformer/mode_platformer.c [hunk 3]
```diff
@@ -1019,7 +232,10 @@ void platformerEnterMode(display_t *disp)
     platformer->prevBtnState = 0;
 
     loadPlatformerHighScores(platformer);
-    //insertScoreIntoHighScores(&(platformer->highScores), 1000000, "EFV", getHighScoreRank(&(platformer->highScores), 1000000));
+    loadPlatformerUnlockables(platformer);
+    if(platformer->highScores.initials[0][0] == 'E' && platformer->highScores.initials[0][1] == 'F' && platformer->highScores.initials[0][2]){
```
Comment (AEFeinstein) [inline main/modes/platformer/mode_platformer.c:236]: Was this supposed to check a third char?

File: main/modes/platformer/mode_platformer.c [hunk 4]
```diff
@@ -1567,27 +1046,69 @@ void updateGameClear(platformer_t *self){
...
+        drawText(d, font, c555, "trip across Swadge Land", 8, 96);
+    }
+    
+    if(gameData->frameCount > 180){
+        drawText(d, font, (gameData->inGameTimer < FAST_TIME) ? cyanColors[(gameData->frameCount >> 3) % 4] : c555, timeStr, 48, 112);
+    }
+
+    if(gameData->frameCount > 300){
+        drawText(d, font, c555, "The Swadge staff", 8, 144);
+        drawText(d, font, c555, "thanks you for playing!", 8, 160);
```
Comment (AEFeinstein) [inline main/modes/platformer/mode_platformer.c:1085]: We sure do!

File: main/modes/platformer/mode_platformer.c [hunk 5]
```diff
@@ -1777,20 +1316,59 @@ void updateShowHighScores(platformer_t *self){
...
+
     self->prevBtnState = self->btnState;
     self->gameData.prevBtnState = self->prevBtnState;
 }
 
 void drawShowHighScores(display_t *d, font_t *font, uint8_t menuState){
-    if(menuState == 3){
-        drawText(d, font, c555, "Your name registrated.", 24, 32);
+    if(platformer->easterEgg){
+        drawText(d, font, highScoreNewEntryColors[(platformer->gameData.frameCount >> 3) % 4], "Happy Birthday, Evelyn!", 20, 32);
```
Comment (AEFeinstein) [inline main/modes/platformer/mode_platformer.c:1334]: ❤

File: main/modes/platformer/mode_platformer.c [hunk 6]
```diff
@@ -1777,20 +1316,59 @@ void updateShowHighScores(platformer_t *self){
...
+    } else if(menuState == 3){
+        drawText(d, font, redColors[(platformer->gameData.frameCount >> 3) % 4], "Your name registrated.", 24, 32);
```
Comment (AEFeinstein) [inline main/modes/platformer/mode_platformer.c:1336]: Registrated? Registered?

**Review by AEFeinstein — CHANGES_REQUESTED** [review]

File: main/modes/platformer/mode_platformer.c (hunk 3 above)
Comment (JVeg199X) [reply main/modes/platformer/mode_platformer.c:236]: Oops, yes. Will fix.

File: main/modes/platformer/platformer_sounds.h (hunk 2 above)
Comment (JVeg199X) [reply main/modes/platformer/platformer_sounds.h:15]: Will do

File: main/modes/platformer/mode_platformer.c (hunk 6 above)
Comment (JVeg199X) [reply main/modes/platformer/mode_platformer.c:1336]: The wording is directly lifted from the Mario Bros. Arcade game. Always thought it was hilarious. I’m keeping it.

File: main/modes/platformer/entity.c (hunk 1 above)
Comment (JVeg199X) [reply main/modes/platformer/entity.c:485]: I need to #define this. 
max uint16_t - (screen height << subpixel resolution)

File: main/modes/platformer/mode_platformer.c (hunk 6 above)
Comment (AEFeinstein) [reply main/modes/platformer/mode_platformer.c:1336]: Thank you so much a for to playing my game

**Review by AEFeinstein — APPROVED** [review]

**Pull request merged**